projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -309,22 +309,21 @@ public Set<String> listJobs() throws ApiException {
    *     Optional if the cron job does not exist or cannot be read
    */
   public Optional<JobDeploymentStatus> readCronJob(String cronJobName) {
-    return readV1CronJob(cronJobName);
-  }
-
-  public Optional<JobDeploymentStatus> readV1CronJob(String cronJobName) {
     log.debug("Reading k8s V1 cron job: {}", cronJobName);
-    V1CronJob cronJob = null;
     try {
-      cronJob = batchV1Api.readNamespacedCronJob(cronJobName, namespace, null);
+      return mapV1CronJobToDeploymentStatus(
+          batchV1Api.readNamespacedCronJob(cronJobName, namespace, null), cronJobName);
     } catch (ApiException e) {
-      log.warn(
-          "Could not read cron job: {}; reason: {}",
-          cronJobName,
-          new KubernetesException("", e).toString());
+      if (e.getCode() == 404) {
+        log.warn(
+            "Could not read cron job: {}; reason: {}",
+            cronJobName,
+            new KubernetesException("", e).toString());
+        return Optional.empty();
+      } else {
+        throw new KubernetesException("", e);
```

Comment:
Nit: Why wrap it if we're not adding a message?

Reply:
Good question. 

KuberentesException in control-service extends ExternalSystemError. 
These are handle explicitly by control service to tell user it is an issue with an external system and not an issue with the control service as such.